### ISSUE TITLE
CloudWatch access with sts:AssumeRoleWithWebIdentity broken in 7.3.x

### ISSUE THREAD

**Issue by uncycler** [grafana/grafana#29017]
**What happened**:

On AWS EKS 1.18, Grafana 7.3.1, deployed with the helm chart, cannot perform sts:AssumeRoleWithWebIdentity authentication anymore using IRSA. Query fails with this error:
failed to call cloudwatch:ListMetrics: WebIdentityErr: failed to retrieve credentials\ncaused by: AccessDenied: Not authorized to perform sts:AssumeRoleWithWebIdentity\n\tstatus code: 403

Cloudwatch datasource provider is set to "AWS SDK Default".

When I change the image back to 7.2.2, using the same chart version, it works as expected.

**What you expected to happen**:

Authenticate succesfully to AWS using IRSA token.

**How to reproduce it (as minimally and precisely as possible)**:

On a EKS 1.18 cluster:
* Create IAM role along with policy that give it access to cloudwatch.
* Deploy Grafana helm chart and set the serviceAccount annotation "eks.amazonaws.com/role-arn" to the IAM role ARN.
* Create a Cloudwatch datasource and test.

**Environment**:
- Grafana version: 7.3.x
- Data source type & version: CloudWatch
- OS Grafana is installed on: Docker
- User OS & Browser: Ubuntu/Chrome

**Comment by sunker** [comment]
@aknuds1 can this be related to the #25548?

**Comment by Legion2** [comment]
We are using AWS EKS IAM Roles for Service Accounts with Grafana 7.3.3. I think the problem here is that you created a new Service Account with a different name and namespace and attached the annotation to it. This does not work, the role can only be attached to a service account with a specific name.

We also deployed grafana with helm and set the name of the service account, after that we created the IAM role with eksctl for that service account.
